zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/tenant/TenantController.java

```diff
@@ -167,6 +183,35 @@ private ResponseEntity<TenantSearchQueryResponse> search(final TenantQuery query
     }
   }
 
+  private ResponseEntity<UserSearchResponse> searchUsersInTenant(
+      final String tenantId, final UserQuery userQuery) {
+    try {
+      final var composedUserQuery = buildUserQuery(tenantId, userQuery);
+      final var result =
+          userServices
+              .withAuthentication(RequestMapper.getAuthentication())
+              .search(composedUserQuery);
+      return ResponseEntity.ok(SearchQueryResponseMapper.toUserSearchQueryResponse(result));
+    } catch (final Exception e) {
+      return mapErrorToResponse(e);
+    }
+  }
+
+  private UserQuery buildUserQuery(final String tenantId, final UserQuery userQuery) {
+    return UserQuery.of(
+        builder ->
+            builder
+                .filter(
+                    f ->
+                        f.name(userQuery.filter().name())
+                            .email(userQuery.filter().email())
+                            .key(userQuery.filter().key())
+                            .username(userQuery.filter().username())
+                            .tenantId(tenantId))
+                .sort(userQuery.sort())
+                .page(userQuery.page()));
```

Comment:
We can use the toBuilder() pattern here now. Maybe we should also assert that the userQuery doesn't contain a tenant id filter already.

Reply:
I wondered about the tenantId on the filter object also but in this case I wanted it to be very intentional that the tenant ID you pass in via the path is the one that we will use moving down the stack (we can also document this in the rest spec). We could add in validation as you say but I wasn't completely convinced in the benefit of blocking the request based on that

Reply:
It points to a problem somewhere downstream and it'd be best for the client if they'd get a 400. We have the same problem with the member-only filters though and maybe this isn't the right layer for this validation.